client/src/app/site/users/services/user-repository.service.ts

```diff
@@ -84,4 +89,70 @@ export class UserRepositoryService extends BaseRepository<ViewUser, User> {
         const groups = this.DS.getMany(Group, user.groups_id);
         return new ViewUser(user, groups);
     }
+
+    /**
+     * Generates a random password
+     *
+     * @param length THe length of the password to generate
+     * @returns a random password
+     */
+    public getRandomPassword(length: number = 8): string {
+        let pw = '';
+        const characters = 'abcdefghijkmnpqrstuvwxyzABCDEFGHJKLMNPQRSTUVWXYZ23456789';
+        for (let i = 0; i < length; i++) {
+            pw += characters.charAt(Math.floor(Math.random() * characters.length));
+        }
+        return pw;
+    }
+
+    public async resetPassword(user: ViewUser, password: string): Promise<void> {
```

Comment:
todo: canmanage or something that only admins might do

Reply:
it is protected with *osPerms in user-list.component.html line 117